### PR TITLE
FIRECERT-2113: [FCS] Process setResponseData in plugins

### PR DESCRIPTION
**Pull request by SnehalP251**

### CODE REVIEW

File: cypress/plugins/testDataProcessor.js [hunk 1]
```diff
@@ -50,16 +55,47 @@ function testDataProcessor(configEnv) {
   ]);
 
   // Resolving the variables in the JSON
-  const resolvedFireboltCallsJson = processFireboltJson(combinedFireboltCallsJson);
+  resolvedFireboltCallsJson = processFireboltJson(combinedFireboltCallsJson);
+
+  // Resolving the variables in the SetResponse JSON
+  const resolvedSetResponseJson = processSetResponseJson(mergedSetResponseJson);
```
Comment (Abhishk123) [inline cypress/plugins/testDataProcessor.js:61]: Inside the set response glue code we are parsing the key name, make sure to remove the parsing from glue code and also check can we able to use the parsed data in beforeOperation(moduleReqId Json)

File: cypress/plugins/testDataProcessor.js [hunk 2]
```diff
@@ -50,16 +55,47 @@ function testDataProcessor(configEnv) {
...
+ *  - Return the JSON with the updated value.
+ *
+ *  @example
+ *  processSetResponseJson({'abc': {'fireboltMock': 'ACKNOWLEDGE_CHALLENGE_GRANTED', 'firstParty': 'true'}})
+ *  @returns
+ *  {'abc': {fireboltMock: {'method': 'method_name', 'response': [{"isCancelled": false, "withUi": true, "result":{}}]}}, 'firstParty': 'true'}
+ */
+function processSetResponseJson(jsonData) {
+  // Looping through json data
+  for (const key in jsonData) {
```
Comment (Abhishk123) [inline cypress/plugins/testDataProcessor.js:85]: if fireboltCall/fireboltMock key value not present returning key as-is right? cross check this changes are okay and inside the setResponse glue if value not parsed we need to handle it like printing the log or failing the test

Comment (Abhishk123) [reply cypress/plugins/testDataProcessor.js:85]: test once it is taking key name as-is when fireboltCall/fireboltMock value not present,

File: cypress/plugins/testDataProcessor.js [hunk 3]
```diff
@@ -50,16 +55,47 @@ function testDataProcessor(configEnv) {
...
+ *  - Iterate over each key in the provided JSON
+ *  - Resolve the values of setResponse params
+ *  - Return the JSON with the updated value.
+ *
+ *  @example
+ *  processSetResponseJson({'abc': {'fireboltMock': 'ACKNOWLEDGE_CHALLENGE_GRANTED', 'firstParty': 'true'}})
+ *  @returns
+ *  {'abc': {fireboltMock: {'method': 'method_name', 'response': [{"isCancelled": false, "withUi": true, "result":{}}]}}, 'firstParty': 'true'}
+ */
+function processSetResponseJson(jsonData) {
```
Comment (Abhishk123) [inline cypress/plugins/testDataProcessor.js:83]: Change "jsonData" to "setResponseJsonData" or different name

Comment (SnehalP251) [reply cypress/plugins/testDataProcessor.js]: Changed "jsonData" to "setResponseJsonData"

File: cypress/plugins/testDataProcessor.js (hunk 2 above)
Comment (SnehalP251) [reply cypress/plugins/testDataProcessor.js]: if fireboltCall/fireboltMock key value is not present, the key will be returned as it is.

File: cypress/plugins/testDataProcessor.js (hunk 1 above)
Comment (SnehalP251) [reply cypress/plugins/testDataProcessor.js:61]: Made changes in beforeOperation to use the parsed data

**Review by jbigel — COMMENTED** [review]
So my understanding of the purpose of this ticket was to move logic to the plugin so we wouldn't have to do as much work during the test execution. I see we are handling that logic in the plugin but I would've expected that would mean we could simplify/remove code that is ran during execution.

Am I missing something?

**Comment by SnehalP251** [comment]
> So my understanding of the purpose of this ticket was to move logic to the plugin so we wouldn't have to do as much work during the test execution. I see we are handling that logic in the plugin but I would've expected that would mean we could simplify/remove code that is ran during execution.
> 
> Am I missing something?

The function parsedMockData added in command.js will return the response based on condition. In beforeOperation we will get the key and for setresponse the response would a processed data. so if object has a key then data would be processed else parsed response will be returned.

**Review by jbigel — APPROVED** [review]

**Pull request merged**